### DISCUSSION TITLE
Compression support for Adonis

### DISCUSSION THREAD

**Discussion by crrmacarse** [adonisjs/core#1664] [General]
Something like compression(https://www.npmjs.com/package/compression) of express. Is this achievable on adonis?

If someone could give me an idea of how to approach this that'll be really great. We're riddled with 1MB+ of frontend assets on the first-page load and I'm really eager on minimizing it.

**Comment by RomainLanz** [comment]
Hey @crrmacarse! 👋 

This is not the job of your AdonisJS application.
This is the job of your webserver (Apache, Nginx, Caddy, etc.)

I recommend you to read some documentation on the same.

▶ https://docs.nginx.com/nginx/admin-guide/web-server/compression/

**Reply by crrmacarse** [reply]
hey @RomainLanz! first off I wanna thank you for your work on adonis. I'm a big fan.

Our project is full stack which means serving frontend assets is handled by NodeJS right? I'm utilizing this approach:

```
Route.any('*', 'ReactViewController.show')
```

where inside `ReactViewController` is a `view.render` method to a compiled assets.

For comparison, in express i could just use:

```
const compression = require('compression');
const express = require('express');

const app = express();
app.use(compression());
```

BONUS: We're also utilizing Heroku and this is in their docs: https://devcenter.heroku.com/articles/compressing-http-messages-with-gzip

**Comment by thetutlage** [comment]
Yeah, I agree folks not using Nginx or any other proxy server may find it useful. I believe, it is just a matter of having a compression middleware. Anyone including you can contribute to the ecosystem by creating the middleware.

**Reply by crrmacarse** [reply]
Noted and thanks @thetutlage. I might try for it 🤞
